### PR TITLE
feat: TET-895 toggle

### PR DESCRIPTION
**Pull request by MartaKozina010**

### CODE REVIEW

File: src/components/Toggle/Toggle.props.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { MouseEvent } from 'react';
+
+import type { ToggleConfig } from './Toggle.styles';
+import { HelperTextProps } from '../HelperText';
+
+export type ToggleProps = {
+  isIndeterminate?: boolean;
+  isChecked?: boolean;
+  size?: 'small' | 'large';
+  state?: 'disabled' | undefined;
```
Comment (mjamrozekvl) [inline src/components/Toggle/Toggle.props.ts:10]: I think we can remove `undefined` because it is already covered by optional property expression.
```suggestion
  state?: 'disabled';
```

File: src/components/Toggle/Toggle.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        isChecked={mainChecked || (toggle1Value && toggle2Value)}
+        onChange={() => setMainChecked(!mainChecked)}
+      />
+      <Toggle
+        isChecked={toggle1Value}
+        onChange={() => setToggle1Value(!toggle1Value)}
+      />
+      <Toggle
+        isChecked={toggle2Value}
+        onChange={() => setToggle2Value(!toggle2Value)}
```
Comment (mjamrozekvl) [inline src/components/Toggle/Toggle.stories.tsx:86]: nitpick: wdyt about using function with previous state as first argument

```suggestion
        onChange={() => setToggle2Value((prevValue) => !prevValue)}
```

File: src/docs-components/ToggleDocs.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                    helperText={
+                      helperText ? { text: 'Helper Text' } : undefined
+                    }
+                    size="large"
+                  />
+                  <Toggle
+                    isIndeterminate={state === 'Indeterminate'}
+                    isChecked={state === 'Checked'}
+                    mt="$dimension-500"
+                    label={label ? 'Label' : (undefined as unknown as string)}
```
Comment (mjamrozekvl) [inline src/docs-components/ToggleDocs.tsx:76]: wdyt about below solution? still compiles properly
```suggestion
                    label={label ? 'Label' : undefined}
```

Comment (mwleklinski) [reply src/docs-components/ToggleDocs.tsx:76]: Yup, that's right. @MartaKozina010 If you ever find yourself having to cast types for Components attributes, that means something is wrong

File: src/theme/theme.ts [hunk 4]
```diff
@@ -1347,12 +1347,12 @@ const fixedTokens = {
     active: '&:active',
     placeholder: '&::placeholder',
     disabled: '&:disabled, &[aria-disabled=true], &[data-state="disabled"]',
-    selected: '&[data-state="selected"]',
+    selected: '&[data-state="selected"], &[aria-selected=true]',
```
Comment (mwleklinski) [inline src/theme/theme.ts:1350]: Why is that necessary here? We should store the selected state in the `data-selected` attribute. @adrian-potepa wdyt

File: src/theme/theme.ts [hunk 5]
```diff
@@ -1347,12 +1347,12 @@ const fixedTokens = {
...
     loading: '&[data-state="loading"]',
     hoverWithoutButton:
       '&:hover:not(:has(button:hover), &:invalid, &[data-state="alert"])',
     alert: '&:invalid, &[data-state="alert"]',
-    indeterminate: '&:indeterminate',
+    indeterminate: '&:indeterminate, &[data-state="indeterminate"]',
```
Comment (mwleklinski) [inline src/theme/theme.ts:1355]: the same issue as above

File: src/docs-components/ToggleDocs.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+                      helperText ? { text: 'Helper Text' } : undefined
+                    }
+                    size="large"
+                  />
+                  <Toggle
+                    isIndeterminate={state === 'Indeterminate'}
+                    isChecked={state === 'Checked'}
+                    state="disabled"
+                    mt="$dimension-500"
+                    label={label ? 'Label' : (undefined as unknown as string)}
```
Comment (mwleklinski) [inline src/docs-components/ToggleDocs.tsx:105]: This casting shouldn't be here. If something is wrong with the label in Toggle, we should fix it

File: src/docs-components/ToggleDocs.tsx [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+                  display="flex"
+                  flexDirection="column"
+                  gap="$space-component-gap-2xLarge"
+                >
+                  <Toggle
+                    isIndeterminate={state === 'Indeterminate'}
+                    isChecked={state === 'Checked'}
+                    mt="$dimension-500"
+                    state="disabled"
+                    label={label ? 'Label' : (undefined as unknown as string)}
```
Comment (mwleklinski) [inline src/docs-components/ToggleDocs.tsx:94]: and here

File: src/docs-components/ToggleDocs.tsx [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+                <tet.div
+                  display="flex"
+                  flexDirection="column"
+                  gap="$space-component-gap-2xLarge"
+                >
+                  <Toggle
+                    isIndeterminate={state === 'Indeterminate'}
+                    isChecked={state === 'Checked'}
+                    mt="$dimension-500"
+                    label={label ? 'Label' : (undefined as unknown as string)}
```
Comment (mwleklinski) [inline src/docs-components/ToggleDocs.tsx:66]: and here

**Review by mwleklinski — CHANGES_REQUESTED** [review]

File: src/components/Toggle/Toggle.props.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  label?: string;
+  helperText?: Pick<HelperTextProps, 'text'>;
+  custom?: ToggleConfig;
+  onChange?: (e?: MouseEvent<HTMLSpanElement>) => void;
+};
```
Comment (adrian-potepa) [inline src/components/Toggle/Toggle.props.ts:15]: `Toggle` should support all the same events and props as `Checkbox` component, its the same business logic, but  different UI ;)

File: src/components/Toggle/index.ts
Comment (adrian-potepa) [inline src/components/Toggle/index.ts:1]: Add `export` in `src/components/index` :)

**Review by adrian-potepa — CHANGES_REQUESTED** [review]
It needs some polishing :)

**Review by adrian-potepa — APPROVED** [review]

**Review by tpiechaczek — APPROVED** [review]
